rd-kt/rd-framework/src/main/kotlin/com/jetbrains/rd/framework/impl/RdTask.kt

```diff
@@ -69,22 +81,15 @@ class CallSiteWiredRdTask<TReq, TRes>(
 
     init {
 
-        call.wire.advise(taskWireSubscriptionDefinition.lifetime, this) //this lifetimeDef listen only one value
+        wire?.advise(taskWireSubscriptionDefinition.lifetime, this) //this lifetimeDef listen only one value
         if (!taskWireSubscriptionDefinition.onTerminationIfAlive { result.setIfEmpty(RdTaskResult.Cancelled()) }) {
             result.setIfEmpty(RdTaskResult.Cancelled())
         }
 
         result.adviseOnce(Lifetime.Eternal) { taskResult ->
             taskWireSubscriptionDefinition.terminate() //no need to listen result or cancellation from wire
 
-            if (taskResult is RdTaskResult.Success && taskResult.value != null && taskResult.value.isBindable()) {
-
-                outerLifetime.executeIfAlive {  // lifetime can be terminated from background thread
-                    outerLifetime.onTermination(::sendCancellation)
-                    taskResult.value.bindPolymorphic(outerLifetime, call, rdid.toString())
-                } ?: sendCancellation()
-
-            } else if (taskResult is RdTaskResult.Cancelled) { //we need to transfer cancellation to the other side
+            if (taskResult is RdTaskResult.Cancelled) { //we need to transfer cancellation to the other side
```

Comment:
Why have we dropped processing of completed tasks here? Is this okay?

Reply:
Because `taskResult` was bound earlier in onWireReceived, this code will be called too late on the passed wire scheduler, but with new behaviour we need to bind result on socket thread, to avoud losing events